File: test/testMovieGenre.js
const movieGenre = require('../js/movieGenre').movieGenre;
const assert = require('assert');

function createMovie(){
    return new Movie();
}

function test_constructor(){
    let m = createMovie();
    assert.deepEqual(m.uuid, "");
}

function test_addRelatedMovie(m){
    let m = createMovie();
    let arm = m.addRelatedMovie(m);
    assert.equal(typeof(arm), 'boolean');
}

function test_setGenre(g){
    let m = createMovie();
    let sg = m.setGenre(g);
    assert.equal(typeof(sg), 'boolean');
}

function createGenre(){
    return new Genre();
}

function test_GConstructor(){
    let g = createGenre();
    assert.deepEqual(g.name, "");
}

function test_addMovie(m){
    let g = createGenre();
    let am = g.addMovie(m);
    assert.equal(typeof(am), 'boolean');
}

const testFunctions = [
    test_constructor,
    test_addRelatedMovie,
    test_setGenre,
    test_GConstructor,
    test_addMovie
].forEach(func =>{
    func.call();
});

File: test/testMovieGenre.py
import sys
sys.path.insert(0, '../py')
from movieGenre import Movie
from movieGenre import Genre
import unittest

class testMovie(unittest.TestCase):
    def __createMovie(self):
        return Movie()

    def test_constructor(self):
        m = self.__createMovie()
        self.assertEqual(m.uuid, "")
    
    def test_addRelatedMovie(self, m):
        m = self.__createMovie()
        arm = m.addRelatedMovie(self, m)
        self.assertEqual(type(arm), bool)
    
    def test_setGenre(self, g):
        m = self.__createMovie()
        sg = m.setGenre(self, g)
        self.assertEqual(type(sg), bool)
    
class testGenre(unittest.TestCase):
    def __createGenre(self):
        return Genre()

    def test_GConstructor(self):
        g = self.__createGenre()
        self.assertEqual(g.name, "")
    
    def test_addMovie(self, m):
        g = self.__createGenre()
        am = g.addMovie(self, m)
        self.assertEqual(type(am), bool)

if __name__ == '__main__':
    unittest.main()
